src/Imgix/UrlHelper.php

```diff
@@ -10,23 +10,34 @@ class UrlHelper {
     private $signKey;
     private $params;
 
-    public function __construct($domain, $path, $scheme = "http", $signKey = "", $params = array(), $rawEncodePath = false) {
+    public function __construct($domain, $path, $scheme = "http", $signKey = "", $params = array()) {
         $this->domain = $domain;
-        $this->path = $this->formatPath( $path, $rawEncodePath );
+        $this->path = $this->formatPath($path);
         $this->scheme = $scheme;
         $this->signKey = $signKey;
         $this->params = $params;
     }
 
-    public function formatPath($path, $rawUrlEncode ) {
+    public function formatPath($path) {
         if (!is_string($path) || strlen($path) < 1)
             return '/';
 
-        if (0 === strpos($path, "http"))
-            $path = $rawUrlEncode ? rawurlencode($path) : urlencode($path);
+        // Strip leading slash first (we'll re-add after encoding)
+        $path = preg_replace("/^\//", "", $path);
 
-        $path = ($path[0] === '/' ? '' : '/') . $path;
-        return $path;
+        if (preg_match("/^https?:\/\//", $path)) {
```

Comment:
Why do you only match on `https`? Could someone also provide a path with `http`?

Reply:
This regex matches both--the `?` following the `s` makes it an optional character.

Reply:
Ah, okay! Got it.

Reply:
I approved the changes, but I think it would still work the same and be more readable to have the regex as `/^http(s)?:\/\//`. But I'll leave that up to you!